aten/src/ATen/native/quantized/cpu/qlinear_dynamic.cpp

```diff
@@ -204,17 +209,186 @@ class QLinearDynamicInt8 final : public torch::OperatorKernel {
 
     return output;
   }
-#else // USE_FBGEMM
-  at::Tensor operator()(
-      at::Tensor /* input */,
-      at::Tensor /* packed_weight */) {
-    // We make a strong guarantee that models using these operators will have
-    // the same numerics across different machines. Therefore, we do not provide
-    // a fallback path and rather fail loudly if we cannot run FBGEMM.
+#endif // USE_FBGEMM
+#ifdef USE_PYTORCH_QNNPACK
+
+  struct qparams {
+    float scale;
+    std::int32_t zero_point;
+  };
+
+  qparams chooseQuantizationParams(
+      float min,
+      float max,
+      int32_t qmin,
+      int32_t qmax) {
+    double scale =
+        (std::max(max, 0.f) - std::min(min, 0.f)) / ((double)qmax - qmin);
+    if (scale == 0) {
+      scale = 0.1;
+    }
+    // If scale is 0, we arbitrary adjust the scale to 0.1
+    assert(scale > 0);
```

Comment:
Use `TORCH_CHECK` instead of `assert`. Also, with the suggestions above, this is unnecessary.